sccp/figures/common.py

```diff
@@ -532,9 +531,14 @@ def plotCellTypeUMAP(points, data, ax):
     """Plots UMAP labeled by cell type"""
     umap.plot.points(points, labels=data["Cell Type"].values, ax=ax)
     
-def plotCmpPerCellType(weightedprojs, cmp, ax):
+def plotCmpPerCellType(weightedprojs, cmp, ax, violins = False):
     """Boxplot of weighted projections for one component across cell types"""
     cmpName = "Cmp. "+str(cmp)
+    if violins == True:
+        sns.violinplot(data=weightedprojs, x=cmpName, y="Cell Type",
+                        inner=None, linewidth=0, ax=ax)
+        for i in range(pd.Series(weightedprojs['Cell Type']).nunique()):
+            ax.collections[i].set_alpha(0.4)
     sns.boxplot(data=weightedprojs[[cmpName, "Cell Type"]], x=cmpName, y="Cell Type", ax=ax)
```

Comment:
thoughts about using something like this @andrewram4287 ? I'm hesitant to completely get rid of the violins since they can display bimodality; although it's true that both distribution visualization options have their pros and cons. Here's what the current image looks like: 
![figureS2b](https://github.com/meyer-lab/scCP/assets/126696611/4510acb0-4d66-4749-8fa9-2657cc4607c1)

Reply:
also let me know if you want me to give up on this pull request and start a new one for this figure since it's gotten a little chaotic trying to merge all of these things 😵‍💫

Reply:
@pascoesean I don't know how much much the violinplots on top of the boxplots tell us more about the data and the variation in these cases. @aarmey What do you think?

Reply:
Yes, violins display bimodality, but that is not the case here. I would get rid of the violins and remove showing the outliers (which are misleading for such a high N).

Reply:
@aarmey Great! One concern about the outliers-- I think for component 13, I was going to make the case that the cells that were most implicated in that component were the megakaryocytes, which is why we see a lot of high outliers across cell types (some of which are presumably megakaryocytes mislabeled as other cell types-- the UMAP region highlighted here was also not one single color by cell type), and this isn't as visible without outliers:
<img width="733" alt="Screenshot 2023-08-11 at 3 33 35 PM" src="https://github.com/meyer-lab/scCP/assets/126696611/2932e31c-a1b9-4fdf-bcda-f5d18e8759d9">

Maybe the answer would be to just refer to the UMAP instead(?)

Reply:
Yes re UMAP. Additionally, long-term we will just need to gate out these cells. In the story, we can start with the box plots, explain that we didn't see anything, but we did see a small localization in the UMAP. From there we can explain how we got to megaKs.